tests/test_lightgbm_converter.py

```diff
@@ -50,45 +50,81 @@ def _run_lgbm_classifier_converter(self, num_classes, extra_config={}):
 
     # Perf_tree_trav classifier
-    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test require LightGBM installed")
+    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test requires LightGBM installed")
     def test_lgbm_perf_tree_trav_classifier_converter(self):
         self._run_lgbm_classifier_converter(2, extra_config={"tree_implementation": "perf_tree_trav"})
 
     # Multi classifier
-    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test require LightGBM installed")
+    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test requires LightGBM installed")
     def test_lgbm_multi_classifier_converter(self):
         self._run_lgbm_classifier_converter(3)
 
     # Gemm multi classifier
-    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test require LightGBM installed")
+    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test requires LightGBM installed")
     def test_lgbm_gemm_multi_classifier_converter(self):
         self._run_lgbm_classifier_converter(3, extra_config={"tree_implementation": "gemm"})
 
     # Tree_trav multi classifier
-    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test require LightGBM installed")
+    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test requires LightGBM installed")
     def test_lgbm_tree_trav_multi_classifier_converter(self):
         self._run_lgbm_classifier_converter(3, extra_config={"tree_implementation": "tree_trav"})
 
     # Perf_tree_trav multi classifier
-    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test require LightGBM installed")
+    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test requires LightGBM installed")
     def test_lgbm_perf_tree_trav_multi_classifier_converter(self):
         self._run_lgbm_classifier_converter(3, extra_config={"tree_implementation": "perf_tree_trav"})
 
+    def _run_lgbm_ranker_converter(self, num_classes, extra_config={}):
+        warnings.filterwarnings("ignore")
+        for max_depth in [1, 3, 8, 10, 12, None]:
+            model = lgb.LGBMRanker(n_estimators=10, max_depth=max_depth)
+            np.random.seed(0)
+            X = np.random.rand(100, 200)
+            X = np.array(X, dtype=np.float32)
+            y = np.random.randint(num_classes, size=100)
+
+            model.fit(X, y, group=[X.shape[0]], eval_set=[(X, y)], eval_group=[X.shape[0]])
+
+            torch_model = hummingbird.ml.convert(model, "torch", extra_config=extra_config)
+            self.assertIsNotNone(torch_model)
+            np.testing.assert_allclose(model.predict(X), torch_model.predict(X), rtol=1e-06, atol=1e-06)
+
+    # Ranker
+    @unittest.skipIf(not lightgbm_installed(), reason="LightGBM test requires LightGBM installed")
+    def test_lgbm_binary_ranker_converter(self):
+        # Using 30 rather than 1000 to avoid "label (31) excel the max range" error
```

Comment:
What is this? Any idea?

Reply:
I followed the instructions to set `label_gain` without luck https://github.com/microsoft/LightGBM/issues/1090

Reply:
I can try again Monday! Maybe there is something minor I was missing

Reply:
Ok let's leave this test as is. Can you add one more with label_gain once you figure it out? I will then merge it!

Reply:
figured out `label_gain=list(range(1000))`. should be ok now i think